pkg/metrics/memory.go

```diff
@@ -58,17 +60,9 @@ func (mm *MemoryMonitor) Sample() (result *MemorySample, err error) {
 		memoryUsedPercent = 100.0 - memoryFreePercent
 	}
 
-	return &MemorySample{
-		MemoryTotal:       float64(memory.Total),
-		MemoryFree:        float64(memory.Available),
-		MemoryUsed:        float64(memory.Used),
-		MemoryCachedBytes: float64(memory.Cached),
-		MemorySlabBytes:   float64(memory.Slab),
-		MemorySharedBytes: float64(memory.Shared),
-
-		MemoryFreePercent: memoryFreePercent,
-		MemoryUsedPercent: memoryUsedPercent,
+	return memorySample(memory, swap, memoryFreePercent, memoryUsedPercent)
+}
 
-		SwapSample: *swap,
-	}, nil
+func floatToReference(value float64) *float64 {
```

Comment:
Is it only used in `memory_linux.go`? If so, I would move it to that file

Reply:
just checked and it's used in darwin too :)